### ISSUE TITLE
Flashing boot for HD.  Don't make my mistakes

### ISSUE THREAD

**Issue by columbusgeek** [EliasKotlyar/Xiaomi-Dafang-Hacks#757]
My first Wazecam I flashed the Open Source U-Boot via [the guide here.](https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/hacks/flashinguboot.md)  It was literally so easy and fast I thougt I did it wrong but everythign verified and works greeat.

I bought another Wyzcam and went through the same steps.  I can into an issue during the "run the following command" step.  I got an error saying:

_"-sh: can't create of=/dev/mtd0: nonexistent directory"_

Weird.  Let's sftp in and see whats in that directory.  
Then the camera clicked and refused to ever boot or show me lights again.

Crapsticks.  That went south quickly.

Welp, I got another one and decided to make extra sure and be very careful.
The exact same thign happend.  

Double crapsticks.

I thought it might be educational to sftp into the first Wyzecam I have to simply see inside to /dev/ via sftp get an idea of what i screwed up.  The second I hit the /dev/ directory via my sftp client the camera goes MIA.  I thougt I broke it too but it eventually comes back online.

So, I think now I have bricked 2 Wyzcams.  Pretty impressive for me, I used to be a iphone Jailbreaker and never shy away from firmware hacks and mods and upgrades and up until now have never borked a device.  Sweet right?  

Is ther something wrong with that command step?  Maybe it needs clarified since i know I'm not the first person to say they screwed up so far.  When the Wyzecam doesn't even show a light when you plug it in, I assume thats a good solid brick right?

Other than this hiccup the Dafang hack is fantastic btw.  Relaly good work.  It's nice to see the software nighttime mode working with the new update as well.

**Comment by jmtatsch** [comment]
With `flash_eraseall /dev/mtd0` you erase your old boot loader.
If you sftp in and something happens (whyever), you are bootloaderless and need to desolder the chip.
Did you maybe have a read-only file system when attempting the bootloader flash?
Before flashing the next boot loader definitely do a` ls /dev/*` before and after the flash_eraseall.

**Comment by columbusgeek** [comment]
> Did you maybe have a read-only file system when attempting the bootloader flash?

Nope.  I had the webgui up looking over things before I did the final change to moving to HD and the new bootloader.  the gui always bleats about things being read-only and I was pleased that with this go-round I wasn't having any of my previous issues.

I know this worked the first time I did it so I must have done this step wrong somehow.  

Do you have to execute both commands at the same time?  For example, I ran:
_flash_eraseall /dev/mtd0_

Then pasted: 
_dd if=<filename.bin> of=/dev/mtd0_ 

and then arrowed back in ssh to edit the filename and inserted the right path to the bin:
_dd if=</system/sdcard/dafang_128mb_v1.bin> of=/dev/mtd0_

**THEN** pressed enter.

Does it all have to be done at the exact same time or things will go south?  

To phrase it another way, should I have pasted this directly in and pressed enter instead of in two steps?

_dd if=<filename.bin> of=/dev/mtd0_ 
_dd if=</system/sdcard/dafang_128mb_v1.bin> of=/dev/mtd0_

**Comment by neozen21** [comment]
I flashed the open source u-boot to all 4 of my WyzeCam V2.. everything went fine.
Once the first one work, I kept using the sdcard for the rest and copy&paste the 2 cmds every time to putty.
The key is to make sure you downloaded the .bin file correctly...
I know some people said they checked the MD5sum, and it still screwed up.. dunno how can that happen though...

**Comment by columbusgeek** [comment]
I noticed here and there I had .conf files get corrupt when trying to clone between sd cards on my Mac.  The content become gibberish.
That was weird but I checked everything before I fully executed it.  

I have a feeling it was somehting to do with the order in which i executed the bootlaod swap.

**Comment by jmtatsch** [comment]
I would run the commands in two separate steps.
I hope you didn't use those brackets <> in your command...

**Comment by columbusgeek** [comment]
> I hope you didn't use those brackets <> in your command...

welp, you just solved the issue.  

I am an idiot.

![facepalm](https://techcrunch.com/wp-content/uploads/2012/06/912accb5_picard-facepalm.png?w=240)

**Comment by columbusgeek** [comment]
Ah, it's all in good fun and it's really no big deal screwing up something that's only 25$ right?  
I grabbed a CH341 to see if I can bring my mistake back from the dead.  

Thank you for updating the doc.  That's the main goal her IMO.  

(btw it should be "Don't do anything stupid **in-between**".)

**Comment by columbusgeek** [comment]
For anybody who is following along, I did a Wyzecam Pan today and the process went fine.

**Comment by ronnyandre** [comment]
Did you manage to unbrick your cameras?
